client/fetch_state.go

```diff
@@ -180,7 +180,7 @@ type fetchTaggedPools interface {
 	MultiReaderIterator() encoding.MultiReaderIteratorPool
 	SeriesIterator() encoding.SeriesIteratorPool
 	MutableSeriesIterators() encoding.MutableSeriesIteratorsPool
-	IteratorArray() encoding.IteratorArrayPool
+	IteratorArray() encoding.MultiReaderIteratorPool
```

Comment:
nit: rename the method name too

Reply:
actually, you can just delete this method - the interface already has `MultiReaderIterator()` returning the type you're after.